### PR TITLE
Docs: add Single Sign-On documentation

### PR DESCRIPTION
**Pull request by mbologna**
Add documentation for the Single Sign-On authentication mechanism.
Targeting Manager-4.0 RC and Uyuni.
PR enabling the functionality: https://github.com/uyuni-project/uyuni/pull/944

Fixes https://github.com/SUSE/spacewalk/issues/7732

### CODE REVIEW

**Comment by Loquacity** [comment]
Added some general wording/grammer/adoc edits.

File: modules/administration/pages/auth-methods.adoc [hunk 1]
```diff
@@ -104,3 +104,78 @@ auth       requisite      pam_vas3.so echo_return
...
+Check your IdP documentation for instructions.
+
+You will need these endpoints:
+
+* Assertion Consumer Service (or ACS): an endpoint to accept SAML messages to establish a session into the Service Provider.
+The endpoint for ACS in {productname} is: https://example.com/rhn/manager/ssaml/acs
+* Single Logout Service (or SLS): an endpoint to initiate a logout request from the IdP.
+The endpoint for SLS in {productname} is: https://example.com/rhn/manager/ssaml/sls
+* Metadata: an endpoint to retrieve {productname} metadata for SAML.
+The endpoint for Metadata in {productname} is https://example.com/rhn/manager/ssaml/metadata
```
Comment (keichwa) [inline modules/administration/pages/auth-methods.adoc]:
```suggestion
The endpoint for Metadata in {productname} is: https://example.com/rhn/manager/ssaml/metadata
```
for consistency

File: modules/administration/pages/auth-methods.adoc [hunk 2]
```diff
@@ -104,3 +104,78 @@ auth       requisite      pam_vas3.so echo_return
...
+
+[IMPORTANT]
+====
+Your IdP must have a SAML:Attribute containing the username of the IdP user domain, called `uid`.
+The `uid` attrribute passed in the SAML:Attribute must be created in the {productname} user base before you activate single sign on.
```
Comment (keichwa) [inline modules/administration/pages/auth-methods.adoc]:
```suggestion
The `uid` attribute passed in the SAML:Attribute must be created in the {productname} user base before you activate single sign on.
```

File: modules/administration/pages/auth-methods.adoc [hunk 3]
```diff
@@ -104,3 +104,78 @@ auth       requisite      pam_vas3.so echo_return
...
+.Procedure: Enabling SSO
+
+. If your users do not yet exist in {productname}, create them first.
+. Edit `/etc/rhn/rhn.conf` and add this line at the end of the file:
++
+----
+java.sson = true
+----
+. Configure `/usr/share/rhn/config-defaults/rhn_java_sson.conf` with the parameters retrieved from your IdP.
+To find all the occurrences you need to change, search in the file for the placeholders [systemitem]``YOUR-PRODUCT`` and [systemitem]```YOUR-IDP-ENTITY``.
```
Comment (keichwa) [inline modules/administration/pages/auth-methods.adoc]:
```suggestion
To find all the occurrences you need to change, search in the file for the placeholders [literal]``YOUR-PRODUCT`` and [literal]```YOUR-IDP-ENTITY``.
```
adoc markup fun

File: modules/administration/pages/auth-methods.adoc [hunk 4]
```diff
@@ -104,3 +104,78 @@ auth       requisite      pam_vas3.so echo_return
...
+Every parameter comes with a brief explanation of what it is meant for.
+. Restart spacewalk-service to pick up the changes:
++
+----
+spacewalk-service restart
+----
+
+When you visit the {productname} URL, you will be redirected to the IdP for SSON where you will be requested to authenticate.
```
Comment (keichwa) [inline modules/administration/pages/auth-methods.adoc]:
```suggestion
When you visit the {productname} URL, you will be redirected to the IdP for SSO where you will be requested to authenticate.
```

**Comment by keichwa** [comment]
At the beginning, you write sign-on (with a dash), but then you continue with writing "sign on" (without a dash).  This need closer checking.  Sometimes it is probably used as a verb and then writing it without a dash is probably fine.

**Comment by mbologna** [comment]
Thanks for all the useful comments. I changed all the occurrences of the term and stayed consistent while using Single Sign-On (SSO). Should be OK now.

**Comment by mbologna** [comment]
Can I have another round of reviews, please? I added/removed some parts and clarified some sections.
As soon as I have some quorum I will merge it. Thanks!

**Pull request merged**